moveit_core/collision_detection/src/world.cpp

```diff
@@ -136,6 +136,51 @@ bool World::hasObject(const std::string& id) const
   return objects_.find(id) != objects_.end();
 }
 
+bool World::knowsTransform(const std::string& id) const
+{
+  // Check object names first
+  if (objects_.find(id) != objects_.end())
+    return true;
+  else  // Then objects' named frames
+  {
+    for (auto o : objects_)
+    {
+      if (o.second->named_frames_.count(id) > 0)
```

Comment:
It's true that every frame name has to be unique, just like TF frames. I will make that clear in the message description.

I don't think this line here is prone to bugs apart from that, though.

Reply:
I think, Mike's comment emphasizes an important implementation issue: While the implementation, using an per-object-map, suggests that we can use the same frame name multiple times (for different objects), actually we can't. It would be better, if this fact would be explicitly reflected in the implementation as well, e.g. using a single global map from frame names to (object, pose) pairs.

Reply:
Hmm, I wouldn't like to rewrite that. It would indeed be much nicer to be able to reuse the same frame name on different objects, but we would have to define how the object namespaces its frames. How is it done in tf2? I think it would be best to stay as close to that convention as possible, so that the frames can (in a final step) be published to tf2 as well.

Reply:
Just for reference, the description I just added to the message:

```
# The names of named frames in the scene need to be unique! If both "screwdriver" and "pencil" have
# a named framed called "tip", it will cause silent errors! Use "screwdriver_tip" and "pencil_tip" instead,
# or submit a pull request to fix this limitation.
```